sql: support readonly default_with_oids var

otan opened the pull request:
This is just one less error when importing PGDUMP.

Release note (sql change): We now support `default_with_oids` which
only accepts being false.

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/74499)
<!-- Reviewable:end -->

rafiss reviewed (approved):
maybe an extra "supported" in the release note, but lgtm!

rafiss commented:
information_schema logictest needs a rewrite i think

otan commented:
done!

bors r=rafiss

otan committed:
sql: support readonly default_with_oids var

This is just one less error when importing PGDUMP.

Release note (sql change): We now support `default_with_oids` which
only accepts being false.

otan commented:
bors r=rafiss
